lib/src/adaptive_gutter.dart

```diff
@@ -0,0 +1,42 @@
+import 'package:flutter/widgets.dart';
+import 'package:flutter_adaptive_scaffold/flutter_adaptive_scaffold.dart';
+
+import '../flutter_gutter.dart';
+
+/// A gutter that switches Gutter size based on the current breakpoint.
+class AdaptiveGutter extends StatelessWidget {
+  /// Creates a new [AdaptiveGutter] widget.
+  const AdaptiveGutter({
+    super.key,
+    this.small = const Gutter.small(),
+    this.medium = const Gutter.medium(),
+    this.large = const Gutter.medium(),
+  });
+
+  /// The type of gutter to create for small screens.
+  final Gutter small;
+
+  /// The type of gutter to create for medium screens.
+  final Gutter medium;
+
+  /// The type of gutter to create for large screens.
+  final Gutter large;
+
+  @override
+  Widget build(BuildContext context) {
+    return SlotLayout(config: <Breakpoint, SlotLayoutConfig>{
+      Breakpoints.small: SlotLayout.from(
```

Comment:
I'm a little confused here.

We have two different axis that affect size:
1. The breakpoint which affects the base definition of a default gutter, defined by Material Design
2. The Gutter variant which defines a power of 2 multiple of the default gutter (some bullshit I made up because it seemed useful)

As far as I can tell, `AdaptiveGutter` is conflating the two, no? Doesn't the `context.materialSpacing` logic already handle this?

FWIW I'm unfamiliar with the slot layout system so I may just need some explanation?

Reply:
No this sets a different Gutter per defined screen size. You would be able to use tiny on a large screen as default.

Reply:
I guess I still don't really get the point of this one.

In my mind `GutterType` controls the scale factor, and screen size (and the corresponding breakpoint) controls the base DP size. Here it looks like we're using screen size to control scale factor AND the underlying logic is still using screen size to control the DP size too? Wouldn't this mean on a large format both the base size and the scale would be bumped up creating an oversized gutter?

But if you're sure this is a coherent use case I'm happy to merge it so as not to waste your time with too much back and forth reviewing-I REALLY appreciate your contribution and don't want to be opinionated on what you add.